test/CMakeLists.txt

```diff
@@ -27,7 +27,7 @@ if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/vol-tests/CMakeLists.txt)
   )
   message(STATUS ${GIT_SUBMODULE_UPDATE_OUTPUT})
 endif()
-add_subdirectory(vol-tests)
+#add_subdirectory(vol-tests)
```

Comment:
This should probably be put behind a CMake option rather than just being disabled as an option

Reply:
The issue this PR was addressing occurs because the vol-tests' configuration depends on HDF5 already being built. Having the vol-tests' CML file invoked like this can only work if a prebuilt library is being used, otherwise I don't think there's a way to tell CMake to wait for HDF5 to get built - as far as I can tell, CMake's `add_custom_target`/dependency logic is for dependencies at _build_ time, not configure time.

That said, the CMake build script still builds the vol-tests from the submodule. This set of changes just means if someone is doing CMake manually, they'll need to invoke it once for the REST VOL/HDF5, and once for the vol-tests. Which isn't perfect, but we'll be using the API tests in the library itself soon anyway.

Reply:
If it isn't currently possible, the VOL tests CMakeLists code could be updated to handle this correctly, but I think the short of it is that we shouldn't be building HDF5 as part of the project in this manner anyway. Probably best at this point to switch over to using [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) or similar.